### PR TITLE
Issue 211: Dockerize

### PR DESCRIPTION
**Pull request by idodin**
## Description
Dockerizes application and implements consequent changes:
* Implements Dockerfile
   * Dockerfile installs Git and sets required Git user config values for HTTPS pulling (used in `update` command)
    * Dockerfile exports poetry requirements and then installs with pip in order to make use of Docker build cache 
    * Dockerfile runs Main.py
* Adds Meta section and Repository variable to config.ini to allow us to specify the exact repository to pull from for `update` command. (n.b. this is required as we must mount our host directory as a volume into the container and as such, the origin remote is set to whatever it is on the host. In order to avoid restricting developers to a single remote protocol (SSH or HTTPS), this allows the update command to pull from whatever branch is active on the host via HTTPS on this repository.)
* Update `update` command to pull via HTTPS strictly. 

## Motivation and Context
See #211 for details on this. This change allows portability across environments and takes us one step closer to fully automating our release process.

## How Has This Been Tested?
Tested most commands locally - tested update repeatedly. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: config/config.ini [hunk 1]
```diff
@@ -18,18 +18,21 @@
 # along with Canary. If not, see <https://www.gnu.org/licenses/>.
 
 [Discord]
-Key = KEY
+Key = KEY
```
Comment (davidlougheed) [inline config/config.ini]: can you remove the space here to kill this diff?

File: config/config.ini [hunk 2]
```diff
@@ -18,18 +18,21 @@
 # along with Canary. If not, see <https://www.gnu.org/licenses/>.
 
 [Discord]
-Key = KEY
+Key = KEY 
 
 [Server]
-ServerID = 236668784948019202
```
Comment (davidlougheed) [inline config/config.ini]: can you reset this?

File: config/config.ini [hunk 3]
```diff
@@ -18,18 +18,21 @@
...
-CommandPrefix = ?
+ServerID = 628008153195347999
+CommandPrefix = !
```
Comment (davidlougheed) [inline config/config.ini]: and this

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.6-slim-buster
+RUN apt update && apt install -y git
+RUN pip install poetry
+RUN git config --global user.name "Martlet"
+RUN git config --global user.email "martlet@gmail.com"
```
Comment (jidicula) [inline Dockerfile]: We have a gmail address for the group: idoneam.collective@gmail.com

**Comment by idodin** [comment]
For reference: freezing to requirements.txt and installing with pip is done to speed up the builds by making use Docker's cache - reference [here](https://github.com/python-poetry/poetry/issues/1301)

**Pull request merged**